js/dragonex.js

```diff
@@ -0,0 +1,912 @@
+                '2': 'entering',
+                '3': 'ok',
+                '4': 'failed',
+            },
+            'withdrawal': {
+                '1': 'pending',
+                '2': 'entering',
+                '3': 'ok',
+                '4': 'failed',
+                '5': 'not approved',
+            },
+        };
+        return (status in statuses[type]) ? statuses[type][status] : status;
+    }
+
+    parseTransaction (transaction, currency = undefined) {
+        const prepayId = this.safeString (transaction, 'prepay_id');
+        const withdrawId = this.safeString (transaction, 'withdraw_id');
+        const id = prepayId !== undefined ? prepayId : withdrawId;
+        const address = this.safeString (transaction, 'addr');
+        let tag = this.safeString (transaction, 'tag');
+        if (tag !== undefined) {
+            if (tag.length < 1) {
+                tag = undefined;
+            }
+        }
+        const txid = this.safeValue (transaction, 'tx_id');
+        const currencyId = this.safeString (transaction, 'coin_id');
+        const coinRes = this.publicGetCoinAll ();
+        const coinList = this.safeValue (coinRes, 'data', []);
+        const coinDict = {};
+        for (let i = 0; i < coinList.length; i++) {
+            coinDict[this.safeString (coinList[i], 'code', '')] = this.safeInteger (coinList[i], 'coin_id', 0);
+        }
+        const code = coinDict[currencyId];
+        let timestamp = undefined;
+        let insertTime = undefined;
+        let applyTime = undefined;
+        if (prepayId !== undefined) {
+            insertTime = this.safeInteger (transaction, 'arrive_time');
+        }
+        if (withdrawId !== undefined) {
+            applyTime = this.safeInteger (transaction, 'arrive_time');
+        }
+        let type = this.safeString (transaction, 'type');
+        if (type === undefined) {
+            if ((insertTime !== undefined) && (applyTime === undefined)) {
+                type = 'deposit';
+                timestamp = insertTime;
+            } else if ((insertTime === undefined) && (applyTime !== undefined)) {
+                type = 'withdrawal';
+                timestamp = applyTime;
+            }
+        }
+        const status = this.parseTransactionStatusByType (this.safeString (transaction, 'status'), type);
+        const amount = this.safeFloat (transaction, 'volume');
+        return {
+            'info': transaction,
+            'id': id,
+            'txid': txid,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'address': address,
+            'tag': tag,
+            'type': type,
+            'amount': amount,
+            'currency': code,
+            'status': status,
+            'updated': undefined,
+            'fee': undefined,
+        };
+    }
+
+    async fetchDepositAddress (code, params = {}) {
+        await this.loadMarkets ();
+        const coinRes = this.publicGetCoinAll ();
```

Comment:
We should implement a unified `fetchCurrencies` method the same way it is implemented in other exchanges, we don't want to load currencies for each and every request to the funding API. That info has to be cached, see how `fetchCurrencies` work everywhere else.